node/src/components/storage.rs

```diff
@@ -252,16 +263,32 @@ impl Storage {
         hard_reset_to_start_of_era: Option<EraId>,
         protocol_version: ProtocolVersion,
         should_check_integrity: bool,
+        network_name: &str,
     ) -> Result<Self, Error> {
         let config = cfg.value();
 
         // Create the database directory.
-        let root = cfg.with_dir(config.path.clone());
-        if !root.exists() {
-            fs::create_dir_all(&root)
-                .map_err(|err| Error::CreateDatabaseDirectory(root.clone(), err))?;
+        let mut root = cfg.with_dir(config.path.clone());
+        let network_subdir = root.join(network_name);
+
+        if !network_subdir.exists() {
+            fs::create_dir_all(&network_subdir)
+                .map_err(|err| Error::CreateDatabaseDirectory(network_subdir.clone(), err))?;
+        }
+
+        let lmdb_files = [
```

Comment:
We normally define things like this as `const` variables towards the top of the file (after the imports).

That would also mean it wouldn't need passed in as an arg to `should_move_storage_files_to_network_subdir()` or `move_storage_files_to_network_subdir()`.